inference-engine/src/hetero_plugin/hetero_executable_network.cpp

```diff
@@ -312,14 +347,377 @@ HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::ICNNNetw
+        std::unordered_map<ngraph::Node*, InputSet> nodeSubgraphCyclicInputDependencies;
+        for (auto&& node : orderedOps) {
+            auto& nodeSubgraphInputDependency = nodeSubgraphInputDependencies[node.get()];
+            auto allNodeSubgraphInputs = Intersection(nodeInputDependencies[node.get()], subgraphInputs);
+            for (auto&& subgraphInput : allNodeSubgraphInputs) {
+                if (subgraphIds[node.get()] == subgraphIds[subgraphInput.get_node()]) {
+                    nodeSubgraphInputDependency.emplace(subgraphInput);
+                }
+            }
+            auto& nodeSubgraphCyclicInputDependency = nodeSubgraphCyclicInputDependencies[node.get()];
+            for (auto&& subgraphInput : allNodeSubgraphInputs) {
+                if (!subgraphInput.get_node()->is_parameter() && subgraphIds[node.get()] == subgraphIds[InputNode(subgraphInput)]) {
+                    nodeSubgraphCyclicInputDependency.emplace(subgraphInput);
+                }
+            }
+        }
+
+        for (auto&& node : orderedOps) {
+            auto& nodeSubgraphCyclicInputDependency = nodeSubgraphCyclicInputDependencies[node.get()];
+            if (!nodeSubgraphCyclicInputDependency.empty()) {
+                auto& nodeSubgraphInputDependency = nodeSubgraphInputDependencies[node.get()];
+                InputSet cyclicInputsDependencies;
+                for (auto&& cyclicInput : nodeSubgraphCyclicInputDependency) {
+                    for (auto&& input : nodeSubgraphInputDependencies[InputNode(cyclicInput)]) {
+                        cyclicInputsDependencies.emplace(input);
+                    }
+                }
+                for (auto&& input : NoConstants(node->inputs())) {
+                    auto& inputNodeSubgraphCyclicInputDependency = nodeSubgraphCyclicInputDependencies[InputNode(input)];
+                    auto& inputNodeSubgraphInputDependency = nodeSubgraphInputDependencies[InputNode(input)];
+                    if (!Intersects(nodeSubgraphCyclicInputDependency,
+                                    inputNodeSubgraphCyclicInputDependency) &&
+                        Intersects(cyclicInputsDependencies, inputNodeSubgraphInputDependency)) {
+                        subgraphInputs.insert(input);
+                    }
+                }
+            }
+        }
+    }
+
+    auto subgraphIds = CollectSubgraphs();
+    // Break graph using insertion of result parameter split
+    NodeMap<ngraph::Node*> subgraphParameterToPrevResult;
+    std::vector<std::shared_ptr<ngraph::op::Result>> results;
+    for (auto&& input : subgraphInputs) {
+        if (!(input.get_node()->is_parameter())) {
+            auto output = input.get_source_output();
+            output.remove_target_input(input);
+            auto result = std::make_shared<ngraph::op::Result>(output);
+            result->set_placement(output.get_node()->get_placement());
```

Comment:
Why you do this?

Reply:
Just copy past from `ngraph::insert_result_parameter_split` function.

Reply:
Should this code be deleted?

Reply:
Yes the placement is a rudiment of nGraph code base and we don't use it in the Inference ENgine. please remove this code.